crates/biome_analyze/src/lib.rs

```diff
@@ -444,28 +446,28 @@ where
                             && suppression.text_range.start() <= start
                     });
 
+            println!("suppression in analyzer xxx: {:?}", suppression);
+
             let suppression = match suppression {
                 Some(suppression) => Some(suppression),
                 None => {
                     let index =
                         self.suppressions
                             .line_suppressions
-                            .binary_search_by(|suppression| {
-                                if suppression.text_range.end() < entry.text_range.start() {
-                                    Ordering::Less
-                                } else if entry.text_range.end() < suppression.text_range.start() {
-                                    Ordering::Greater
-                                } else {
-                                    Ordering::Equal
-                                }
+                            .partition_point(|suppression| {
+                                suppression.text_range.end() < entry.text_range.start()
```

Comment:
based on the [Rust documentation for partition_point](https://doc.rust-lang.org/std/primitive.slice.html#method.partition_point), is my understanding correct that the order `suppression.text_range.end() < entry.text_range.start()` follows Rust conventions for finding the boundary point of non-overlapping ranges?

Reply:
yes, i think this will be the right way here

Reply:
thanks